### PR TITLE
Filter continueConversation events when logging

### LINKED ISSUE #2805: TranscriptLoggerMiddleware is logging ContinueConversation events to transcript log (JS)
See [parent](https://github.com/microsoft/botframework-sdk/issues/6059)

### PR DESCRIPTION
**Pull request by joshgummersall**
Fixes #2805

This PR also includes a small test refactoring to use `sinon` for mocking and verifying error handling. It's a more natural way to set up expectations and verify them without patching methods. It was a bit of bikeshedding, but it should serve as a useful example for future testing scenarios.

### CODE REVIEW

File: libraries/botbuilder-core/tests/transcriptMiddleware.test.js [hunk 1]
```diff
@@ -316,81 +298,80 @@ describe(`TranscriptLoggerMiddleware`, function () {
...
-            }
-        }
-
-        it(`should print to console errors from synchronous logActivity()`, function (done) {
-            stubConsoleError(done, undefined, 2);
-            var adapter = new TestAdapter(async (context) => {
-            }).use(new TranscriptLoggerMiddleware(new SyncErrorLogger()));
-    
-            adapter.send('test');
+    describe("'s error handling", function () {
```
Comment (joshgummersall) [inline libraries/botbuilder-core/tests/transcriptMiddleware.test.js:313]: For reference, this is the major test refactor portion.

File: libraries/botbuilder-core/src/transcriptLogger.ts [hunk 2]
```diff
@@ -47,12 +53,12 @@ export class TranscriptLoggerMiddleware implements Middleware {
 
         // hook up onSend pipeline
         context.onSendActivities(
-            async (ctx: TurnContext, activities: Partial<Activity>[], next2: () => Promise<ResourceResponse[]>) => {
+            async (ctx: TurnContext, activities: Partial<Activity>[], next: () => Promise<ResourceResponse[]>) => {
```
Comment (stevengum) [inline libraries/botbuilder-core/src/transcriptLogger.ts:56]: These random `next2` have caused me random bits of anguish, thanks for changing the name!

**Review by stevengum — CHANGES_REQUESTED** [review]
For the actual values of constants, enums, etc we do stay aligned with what's used in C#/other languages. Once the .NET PR lands the ActivityEventNames enum in JS should match. Other than that, the PR looks solid!

As a general rule, if it's a value that might be serialized and could be shared across languages, it should be aligned across all languages.

File: libraries/botframework-schema/src/index.ts [hunk 3]
```diff
@@ -1725,6 +1725,17 @@ export enum RoleTypes {
...
 
+/**
+ * Defines values for ActivityEventNames.
+ * Possible values include: 'continueConversation', 'createConversation'
+ * @readonly
+ * @enum {string}
+ */
+export enum ActivityEventNames {
+  ContinueConversation = 'continueConversation',
+  CreateConversation = 'createConversation'
```
[the remark this answers is not in the document]
Comment (joshgummersall) [reply libraries/botframework-schema/src/index.ts:1736]: Good catch - that makes sense.

**Review by stevengum — APPROVED** [review]
LGTM!

**Pull request merged**